src/script/interpreter.cpp

```diff
@@ -1452,11 +1534,25 @@ void PrecomputedTransactionData::Init(const T& txTo, std::vector<CTxOut>&& spent
         if (uses_bip341_taproot && uses_bip143_segwit) break; // No need to scan further if we already need all.
     }
 
-    if (uses_bip143_segwit || uses_bip341_taproot) {
+    if (uses_bip143_segwit || uses_bip341_taproot || uses_bip119_ctv) {
         // Computations shared between both sighash schemes.
         m_prevouts_single_hash = GetPrevoutsSHA256(txTo);
         m_sequences_single_hash = GetSequencesSHA256(txTo);
         m_outputs_single_hash = GetOutputsSHA256(txTo);
+
+        bool skip_scriptSigs = std::find_if(txTo.vin.begin(), txTo.vin.end(),
+                [](const CTxIn& c) { return c.scriptSig != CScript(); }) == txTo.vin.end();
+        if (skip_scriptSigs) {
+            // 0 hash used to signal if we should skip scriptSigs
+            // when re-computing for different indexes.
+            m_scriptSigs_single_hash = uint256{};
+            // TODO: Cache midstate?
+            m_standard_template_single_hash = GetStandardTemplateHashEmptyScript(txTo, m_outputs_single_hash, m_sequences_single_hash, 0);
+        } else {
+            m_scriptSigs_single_hash = GetScriptSigsSHA256(txTo);
+            m_standard_template_single_hash = GetStandardTemplateHashWithScript(txTo, m_outputs_single_hash, m_sequences_single_hash, m_scriptSigs_single_hash, 0);
+        }
+        m_bip119_ctv_ready = true;
```

Comment:
changing this to false has `feature_checktemplateverify` still pass

Reply:
Yep, that's normal.

If the cache is disabled (which making ready = false does), then the cache isn't used.

Reply:
> If the cache is disabled (which making ready = false does), then the cache isn't used.

Just observing that this does not apply anymore in the current version, because Commit 4c823859fc2e46606d13eef8d21ce7aa1fe3edea ("Use HandleMissingData for unexpected missing precomputed data" ) introduces a strict dependence on the cached data, resulting in an assert if the data wasn't cached.

Reply:
great point and solid review note, was a part of rebasing this onto the new HandleMissingData check, which changed this stuff to errors to be missing across the codebase.

Note that it's still possible the CTV hash is not cached, but has to be computed from the other cached parts.

Reply:
btw if curious, https://github.com/bitcoin/bitcoin/pull/21330

it's mostly for signing analysis, not consensus, that missing data paths were every supported. the referenced pr makes it a bit more clean.